api/app/Services/OpenIdBearerTokenService.php

```diff
@@ -107,9 +64,34 @@ private function getConfiguration(string $keyId) : ?Configuration
             return Http::get($jwks_uri)->body();
         });
 
-        $keyDictionary = $this->parseJwksJsonString($jsonString);
+        // Uses web-token/jwt-core to generate public key from "e" and "n" in JWKS.
+        // Source: https://github.com/lcobucci/jwt/issues/32#issuecomment-907556410
+        $set = JWKSet::createFromKeyData(json_decode($jsonString, true));
+        $jwk = $set->get($keyId);
+
+        switch ($jwk->get('alg')) {
+            case 'RS256':
+                $signer = new Signer\Rsa\Sha256();
+                break;
+            case 'RS384':
+                $signer = new Signer\Rsa\Sha384();
+                break;
+            case 'RS512':
+                $signer = new Signer\Rsa\Sha512();
+                break;
+            default:
+                throw new Exception('Unknown algorithm type in jwks');
+        }
+
+        $pem = RSAKey::createFromJWK($jwk)->toPEM();
+        $config = Configuration::forAsymmetricSigner(
+            $signer,
+            InMemory::empty(), // Private key is only used for generating tokens, which is not being done here, therefore empty is used.
+            InMemory::plainText($pem),
+        );
+
+        return $config;
```

Comment:
There's some other Laravel Auth stuff left over here that can be removed:
- `api/app/Services/LocalAuthBearerTokenService.php`
- `api/tests/Unit/LocalAuthBearerTokenTest.php`

Additionally, this file can have the branching logic removed: `api/app/Providers/BearerTokenServiceProvider.php` which would allow you to get rid of `api/app/Services/Contracts/BearerTokenServiceInterface.php` entirely.

Reply:
I might need a little guidance with pulling out the interface. Might you be able to offer a short diff I removing it in one place? (assuming that's easier than typing out an explanation or hopping on a quick call, both of which I'd also be grateful for!)

Reply:
Yeah, good chance to do some pairing. :wink: Let's look at it today sometime.